core/main/src/firebolt/handlers/localization_rpc.rs

```diff
@@ -125,6 +129,61 @@ pub trait Localization {
     ) -> RpcResult<ListenerResponse>;
 }
 
+enum MapEntryProperty {
+    Set(SetMapEntryProperty),
+    Remove(RemoveMapEntryProperty),
+}
+
+async fn update_additional_info(
+    mut platform_state: PlatformState,
+    map_entry_property: MapEntryProperty,
+) -> RpcResult<()> {
+    match BrokerUtils::process_internal_main_request(
+        &mut platform_state,
+        "localization.additionalInfo",
+        None,
+    )
+    .await
+    {
+        Ok(Value::Object(additional_info_map)) => {
+            let mut info_map = additional_info_map.clone();
```

Comment:
You may avoid cloning by using mutable `additional_info_map`.

`Ok(Value::Object(mut additional_info_map))`